shell/common/isolate_configuration.cc

```diff
@@ -88,6 +88,35 @@ class SourceIsolateConfiguration final : public IsolateConfiguration {
   FXL_DISALLOW_COPY_AND_ASSIGN(SourceIsolateConfiguration);
 };
 
+class KernelListIsolateConfiguration final : public IsolateConfiguration {
```

Comment:
Should the thing that consumes an IsolateConfiguration be taught about partial loads? Maybe IsolateConfiguration needs a bit that tells its consumer whether to call `DoPrepareIsolate(isolate)`, or `DoPrepareIsolateWithPiece(isolate, piece, /* last */ false)`.

Reply:
It could be hidden in the implementation of an IsolateConfiguration by blocking until all the loads complete. The caller can't proceed until loading is complete anyway.